main/main.c

```diff
@@ -102,6 +140,72 @@ rte_strdup(const char *type, const char *s)
 	return strcpy(res, s);
 }
 
+FILE *log_file = NULL;
+struct rte_timer log_timer;
+
+static int
+cleanup_log(void)
+{
+	if (log_file != NULL) {
+		fclose(log_file);
+		log_file = NULL;
+	}
+
+	rte_free(log_base_dir);
+	rte_free(log_file_name_format);
+
+	return 0;
+}
+
+int
+gatekeeper_log_init(void)
+{
+	int ret;
+	time_t now;
+	struct tm *time_info;
+	char log_file_name[100];
+	char log_file_path[100];
+	FILE *new_log_file = NULL;
+
+	/*
+	 * Initialize log_file with stderr to guarantee that log_file
+	 * has a valid value even before a file is open.
+	 */
+	log_file = stderr;
+
+	time(&now);
+	time_info = localtime(&now);
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-nonliteral"
+	ret = strftime(log_file_name, sizeof(log_file_name),
+		log_file_name_format, time_info);
+#pragma GCC diagnostic pop
+	RTE_VERIFY(ret > 0 && ret < (int)sizeof(log_file_name));
+
+	ret = snprintf(log_file_path, sizeof(log_file_path),
+		"%s/%s", log_base_dir, log_file_name);
+	RTE_VERIFY(ret > 0 && ret < (int)sizeof(log_file_path));
+
+	new_log_file = fopen(log_file_path, "a");
+	if (new_log_file == NULL) {
+		RTE_LOG(ERR, GATEKEEPER,
+			"gatekeeper: Failed to open log file %s - %s\n",
+			log_file_path, strerror(errno));
+		return -1;
+	}
+
+	ret = rte_openlog_stream(new_log_file);
+	if (ret != 0)
+		return -1;
+
+	if (log_file != NULL)
+		fclose(log_file);
```

Comment:
Bug: we are closing `stderr` here. Make sure that `log_file` is not `stderr` before closing it.